### PR TITLE
feat: Added --versions option to the code generator

### PR DESCRIPTION
**Pull request by jkaster**
This change resulted in small but low-level changes that had a major ripple effect through the Typescript codebase

The `sdk-codegen-*` packages are now also marked as deprecated.

### CODE REVIEW

**Review by joeldodge79 — APPROVED** [review]

File: packages/extension-api-explorer/src/ExtensionApiExplorer.tsx [hunk 1]
```diff
@@ -25,19 +25,21 @@
...
+import {
+  ApiModel,
+  getSpecsFromVersions,
+  SpecItem,
+  SpecList,
+  upgradeSpecObject,
+} from '@looker/sdk-codegen'
 import { Looker31SDK, Looker40SDK } from '@looker/sdk'
-import { getSpecsFromVersions } from '@looker/api-explorer/src/reducers'
+import ApiExplorer from '@looker/api-explorer/lib/ApiExplorer'
```
Comment (bryans99) [inline packages/extension-api-explorer/src/ExtensionApiExplorer.tsx:42]: NOTE TO SELF: Need to figure out why we have to reach in to get ApiExplorer. Unrelated to this change.

File: packages/sdk-codegen-utils/README.md [hunk 2]
```diff
@@ -2,6 +2,11 @@
 
 This package contains utility routines for the SDK code generator projects
 
-The code in this package is really trivial and probably not pulling in as a dependency to anything.
+The code in this package is really trivial and should not be used directly in your own projects.
+
+This package has no node dependencies.
+
+**DEPRECATED AND NOT SUPPORTED**: Looker does not support direct use of this package, other than via direct use in the
```
Comment (bryans99) [inline packages/sdk-codegen-utils/README.md:9]: other than via direct use in the
should be?
other than via indirect use in the

File: packages/sdk-codegen/README.md [hunk 3]
```diff
@@ -2,6 +2,11 @@
 
 This package contains the OpenAPI analysis files, and all source code necessary to generate method and type declarations for supported languages.
 
+This package can be used in a browser because it has no node dependencies.
+
+**DEPRECATED AND NOT SUPPORTED**: Looker does not support direct use of this package, other than via direct use in the
```
Comment (bryans99) [inline packages/sdk-codegen/README.md:7]: via indirect use?

**Review by bryans99 — APPROVED** [review]
LGTM

Possibly a couple of typos in README.

File: packages/sdk-codegen/README.md (hunk 3 above)
Comment (jkaster) [reply packages/sdk-codegen/README.md:7]: Fixed in other PR. Thanks!

File: packages/sdk-codegen/src/specConverter.ts [hunk 4]
```diff
@@ -26,7 +26,8 @@
 
 import { isEmpty } from 'lodash'
 import { APIMethods } from '@looker/sdk-rtl'
-import { ApiModel, ArgValues, IApiModel } from './sdkModels'
+import { IApiVersion, IApiVersionElement } from '@looker/sdk'
```
Comment (joeldodge79) [inline packages/sdk-codegen/src/specConverter.ts:29]: seems a little counter-direction to be importing parts of our looker specific API spec here, no? Is there a generic/standard version/schema for this defined somewhere? If not, I'd suggest we define our own local copy here even if it's an exact replica of the Looker interfaces - that way this package isn't relying on `@looker/sdk`

File: packages/sdk-codegen/package.json [hunk 5]
```diff
@@ -33,9 +33,10 @@
     "watch": "yarn lerna exec --scope @looker/sdk-codegen --stream 'BABEL_ENV=build babel src --root-mode upward --out-dir lib --source-maps --extensions .ts,.tsx --no-comments --watch'"
   },
   "dependencies": {
-    "@looker/sdk-codegen-utils": "^21.0.9",
-    "@looker/sdk-node": "^21.0.9",
-    "@looker/sdk-rtl": "^21.0.9",
+    "@looker/sdk": "^21.0.10",
```
Comment (joeldodge79) [inline packages/sdk-codegen/package.json:36]: see my other comment about hopefully dropping this dependency

File: packages/sdk-codegen/package.json [hunk 6]
```diff
@@ -33,9 +33,10 @@
...
+    "@looker/sdk-codegen-utils": "^21.0.10",
+    "@looker/sdk-node": "^21.0.10",
```
Comment (joeldodge79) [inline packages/sdk-codegen/package.json:38]: we should move this to `devDependencies`

**Review by joeldodge79 — COMMENTED** [review]
just some observations about small things we may want to clean up at some point

File: packages/sdk-codegen/package.json (hunk 5 above)
Comment (jkaster) [reply packages/sdk-codegen/package.json:36]: Yup. Let's put up a new PR for that. Drop `@looker/sdk` and remove it from sdk-codegen imports and see where we're at!

File: packages/sdk-codegen/src/specConverter.ts (hunk 4 above)
Comment (jkaster) [reply packages/sdk-codegen/src/specConverter.ts:29]: Yeah, I think I actually did that duplication in another source file. Reliance on `@looker/sdk-rtl` is fine, but it shouldn't rely at all on `@looker/sdk`. Then we can document that "versions" payload as something people could use to generate their own specs.

File: packages/sdk-codegen/package.json (hunk 6 above)
Comment (jkaster) [reply packages/sdk-codegen/package.json:38]: Remove `@looker/sdk` and `@looker/sdk-node`. We should only need `@looker/sdk-rtl` hopefully. The `sdk-codegen-utils` package is so incredibly trivial (one small source file) we could just duplicate it for `sdk-codegen` and `sdk-codegen-scripts`.

Comment (jkaster) [reply packages/sdk-codegen/package.json:38]: I captured this in #533

**Pull request merged**